test/common/event/dispatcher_impl_test.cc

```diff
@@ -482,6 +491,102 @@ TEST_F(DispatcherImplTest, RunPostCallbacksLocking) {
   }
 }
 
+TEST_F(DispatcherImplTest, DispatcherThreadDeleted) {
+  dispatcher_->deleteInDispatcherThread(std::make_unique<TestDispatcherThreadDeletable>(
+      [this, id = api_->threadFactory().currentThreadId()]() {
+        ASSERT(id != api_->threadFactory().currentThreadId());
+        {
+          Thread::LockGuard lock(mu_);
+          ASSERT(!work_finished_);
+          work_finished_ = true;
+        }
+        cv_.notifyOne();
+      }));
+
+  Thread::LockGuard lock(mu_);
+  while (!work_finished_) {
+    cv_.wait(mu_);
+  }
+}
+
+TEST_F(DispatcherImplTest, DispatcherThreadDeletedAtNextCycle) {
```

Comment:
The body of this test is exactly the same as DispatcherImplTest.DispatcherThreadDeleted

Either make them different or remove the duplicate.

Reply:
rewrite: add order enforcement and schedule 2 delete